libsolidity/codegen/ABIFunctions.cpp

```diff
@@ -1238,26 +1246,29 @@ string ABIFunctions::abiDecodingFunctionCalldataArray(ArrayType const& _type)
 					arrayPos := add(offset, 0x20)
 					if gt(add(arrayPos, mul(length, <stride>)), end) { <revertStringPos> }
 				}
-			)";
+			)");
+			for (auto& parameter: parameters)
+				w(parameter.first, parameter.second);
+			w("revertStringOffset", revertReasonIfDebug("ABI decoding: invalid calldata array offset"));
+			w("revertStringLength", revertReasonIfDebug("ABI decoding: invalid calldata array length"));
+			result = w.render();
+		}
 		else
-			templ = R"(
+		{
+			Whiskers w(R"(
 				// <readableTypeName>
 				function <functionName>(offset, end) -> arrayPos {
 					arrayPos := offset
 					if gt(add(arrayPos, mul(<length>, <stride>)), end) { <revertStringPos> }
 				}
-			)";
-		Whiskers w{templ};
-		// TODO add test
-		w("revertStringOffset", revertReasonIfDebug("ABI decoding: invalid calldata array offset"));
-		w("revertStringLength", revertReasonIfDebug("ABI decoding: invalid calldata array length"));
-		w("revertStringPos", revertReasonIfDebug("ABI decoding: invalid calldata array stride"));
-		w("functionName", functionName);
-		w("readableTypeName", _type.toString(true));
-		w("stride", toCompactHexWithPrefix(_type.calldataStride()));
-		if (!_type.isDynamicallySized())
+			)");
+			for (auto& parameter: parameters)
+				w(parameter.first, parameter.second);
```

Comment:
Couldn't this loop and the render call be moved out of the branches?

Reply:
Yes, sure. I introduced now `Whiskers::setTemplate`, with this the same instance can be used.